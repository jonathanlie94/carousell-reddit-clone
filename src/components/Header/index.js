/*
  We can consider moving Header into containers in case we want to make
  a more complicated interaction on the Header that requires connect to our reducers.
*/

import React, { Component } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: 100%;
  height: auto;
`;

class Header extends Component {
  render() {
    return (
      <Wrapper>
        {this.props.children}
      </Wrapper>
    );
  }
}

export default Header;
