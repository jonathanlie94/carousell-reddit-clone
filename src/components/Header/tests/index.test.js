import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import Header from '../';

describe('<Header />', () => {
  it('should render children when passed in', () => {
    const renderedComponent = shallow(
      <Header>
        <div className="unique-class-name" />
      </Header>
    );
    expect(
      renderedComponent.contains(<div className="unique-class-name" />)
    ).toEqual(true);
  });

  it('should render a div', () => {
    const header = shallow(<Header />).dive();
    expect(header.type()).toEqual('div');
  });

  it('should match previous snapshot', () => {
    const header = shallow(<Header />).dive();
    const tree = toJSON(header);
    expect(tree).toMatchStyledComponentsSnapshot();
  });
});
