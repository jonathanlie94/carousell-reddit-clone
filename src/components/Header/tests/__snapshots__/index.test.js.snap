// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match previous snapshot 1`] = `
.ctpUMM {
  width: 100%;
  height: auto;
}

<div
  className="sc-bdVaJa ctpUMM"
/>
`;
